internal/scm/bitbucketserver/bitbucket_server.go

```diff
@@ -130,7 +131,8 @@ func (b *BitbucketServer) GetRepositories(ctx context.Context) ([]scm.Repository
 	for _, bitbucketRepository := range bitbucketRepositories {
 		response, getDefaultBranchErr := client.DefaultApi.GetDefaultBranch(bitbucketRepository.Project.Key, bitbucketRepository.Slug)
 		if getDefaultBranchErr != nil {
-			return nil, getDefaultBranchErr
```

Comment:
I would greatly prefer if the error was checked to confirm this is an error because not branch actually exist (ideally the GetDefaultBranch API should not error but return a bool or set the returned value to empty string).

With this change we might sometimes skip repos semi-silently if there is something like a network error.